File: common/src/main/java/org/tdf/common/trie/ScanValues.java
package org.tdf.common.trie;

import lombok.AccessLevel;
import lombok.Getter;
import org.tdf.common.util.ByteArraySet;

@Getter(AccessLevel.PACKAGE)
class ScanValues implements ScannerAction {
    private ByteArraySet bytes = new ByteArraySet();

    @Override
    public void accept(TrieKey path, Node node) {
        if (node.getType() != Node.Type.EXTENSION && node.getValue() != null) {
            bytes.add(node.getValue());
        }
    }
}

File: common/src/main/java/org/tdf/common/trie/ScanValues.kt
package org.tdf.common.trie

import org.tdf.common.util.ByteArraySet

internal class ScanValues : ScannerAction {
    val bytes = ByteArraySet()
    override fun scan(path: TrieKey, node: Node): Boolean {
        if (node.type != Node.Type.EXTENSION && node.value != null) {
            bytes.add(node.value)
        }
        return true
    }
}